utils.go

```diff
@@ -82,17 +83,39 @@ func GetRootlessRuntimeDir(rootlessUID int) (string, error) {
 	return path, nil
 }
 
+func getInitCommand() (string, error) {
+	file, err := os.Open("/proc/1/comm")
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+	buf, err := ioutil.ReadAll(file)
+	if err != nil {
+		return "", err
+	}
+	return string(buf), nil
+}
+
 func getRootlessRuntimeDir(rootlessUID int) (string, error) {
 	runtimeDir, err := homedir.GetRuntimeDir()
 	if err == nil {
 		return runtimeDir, nil
 	}
-	tmpDir := fmt.Sprintf("/run/user/%d", rootlessUID)
+	initCommand, err := getInitCommand()
+	var tmpDir string
+	if err == nil && initCommand == "systemd" {
+		// Systemd maintains lingering mode for /run/user/X.
+		// It is up to user whether this directory will survive logout.
+		tmpDir = fmt.Sprintf("/run/user/%d", rootlessUID)
+	} else {
+		// Other systems follows standard https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html.
+		// /run/user/X can't survive logout, so it is possible to use temp directory only.
+		tmpDir = fmt.Sprintf("%s/storage-user-%d", os.TempDir(), rootlessUID)
+	}
```

Comment:
we already have a fallback to TempDir so please drop the else branch completely.

If you'd like to change the directory name, please change the existing `tmpDir = fmt.Sprintf("%s/%d", os.TempDir(), rootlessUID)` to `tmpDir = fmt.Sprintf("%s/containers-user-%d", os.TempDir(), rootlessUID)` (let's use containers instead of storage).

Reply:
done